miio/huizuo.py

```diff
@@ -147,7 +311,288 @@ def set_brightness(self, level):
+            "\n"
+            "------------Fan parameters -------------\n"
+            "Fan power:  {result.is_fan_on}\n"
+            "Fan level: {result.fan_speed_level}\n"
+            "Fan mode: {result.fan_mode}\n"
+            "Fan reverse: {result.is_fan_reverse}\n"
+            "\n",
+        ),
+    )
+    def status(self) -> HuizuoStatus:
+        """Retrieve properties."""
+
+        return HuizuoStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    # Fan Reverse option is not available for all models with fan
+    @command(
+        default_output=format_output("Enable fan reverse"),
+    )
+    def fan_reverse_on(self):
+        """Enable fan reverse (only for models which support this fan option)"""
+        if self.model in MODELS_WITH_FAN_WY:
+            return self.set_property("fan_motor_reverse", True)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Disable fan reverse"),
+    )
+    def fan_reverse_off(self):
+        """Disable fan reverse (only for models which support this fan option)"""
+        if self.model in MODELS_WITH_FAN_WY:
+            return self.set_property("fan_motor_reverse", False)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+
+class HuizuoLampHeater(Huizuo):
+    """Support for Huizuo Lamps with heater
+
+    The next section contains the heater management commands
+    Right now I have no devices with the heater for live testing, so the following section
+    generated based on device specitifations"""
+
+    @command(
+        default_output=format_output("Heater powering on"),
+    )
+    def heater_on(self):
+        """Power heater on (only for models with heater)."""
+        if self.model in MODELS_WITH_HEATER:
+            return self.set_property("heater_power", True)
+        else:
+            raise HuizuoException("Your device doesn't support a heater management")
+
+    @command(
+        default_output=format_output("Heater powering off"),
+    )
+    def heater_off(self):
+        """Power heater off (only for models with heater)."""
+        if self.model in MODELS_WITH_HEATER:
+            return self.set_property("heater_power", False)
+        else:
+            raise HuizuoException("Your device doesn't support a heater management")
+
+    @command(
+        click.argument("heat_level", type=int),
+        default_output=format_output("Setting heat level to {heat_level}"),
+    )
+    def set_heat_level(self, heat_level):
+        """Set heat level (only for models with heater)"""
+        if self.model in MODELS_WITH_HEATER:
+            if heat_level not in [1, 2, 3]:
+                raise HuizuoException("Invalid heat level: %s" % heat_level)
```

Comment:
See my previous comment on indents & failing early (so move this before the model check).

Reply:
Moved